java/code/src/com/redhat/rhn/common/hibernate/HibernateFactory.java

```diff
@@ -133,6 +135,108 @@ public static void registerComponentName(String componentName) {
      */
     protected abstract Logger getLogger();
 
+    /**
+     * Finds a single instance of a persistent object, given one parameter of type long
+     * This convenience method is aimed at replacing simple named queries on a single parameter
+     * e.g. an environment like:
+     *
+     * ActionType-xbm.xml:
+     *      <query name="ActionType.findById">
+     *         <![CDATA[from com.redhat.rhn.domain.action.ActionType as t where t.id = :id]]>
+     *     </query>
+     * Anywhere in the code:
+     *     ActionType actType =
+     *          hibernateFactory.lookupObjectByNamedQuery("ActionType.findById", Map.of("id", id), true);
+     *
+     * can be substituted with
+     *
+     * ActionType-xbm.xml:
+     *      .... removed! .....
+     * Anywhere in the code:
+     *      ActionType actType = hibernateFactory.lookupObjectByParam("id", id, ActionType.class, true);
+     *
+     * @param paramName the parameter name (e.g. "label", "name", "id" etc.)
+     * @param paramValue the parameter actual value
+     * @param objClass class of the query object
+     * @param cacheable if we should cache the results of this object
+     * @return Instance of the found object by the query or null if nothing is found.
+     */
+    protected <T> T lookupObjectByParam(String paramName, Long paramValue, Class<T> objClass, boolean cacheable) {
+        String sanitizedParamName = paramName.replaceAll("[^A-Za-z]+", "");
+
+        try {
+            Session session = HibernateFactory.getSession();
+            String query = "FROM %s WHERE %s = :param".formatted(objClass.getSimpleName(), sanitizedParamName);
+            return session.createQuery(query, objClass) //NOSONAR dynamically formatted SQL query is safe here
+                    .setParameter("param", paramValue, LongType.INSTANCE)
+                    .setCacheable(cacheable)
+                    .uniqueResult();
+        }
+        catch (HibernateException he) {
+            throw new HibernateRuntimeException("lookupObjectByParam failed with param [%s]=[%d] on class [%s]"
+                    .formatted(paramName, paramValue, objClass.getSimpleName()), he);
+        }
+    }
+
+    protected <T> T lookupObjectByParam(String paramName, Long paramValue, Class<T> objClass) {
+        return lookupObjectByParam(paramName, paramValue, objClass, false);
+    }
+
+    protected <T> T lookupObjectByParamCacheable(String paramName, Long paramValue, Class<T> objClass) {
+        return lookupObjectByParam(paramName, paramValue, objClass, true);
+    }
```

Comment:
Maybe `cacheable` should be `true` by default? Do you have already an idea on what is the most common value in our cases? 

Also I would avoid having a separate overload for the other case, calling `singleton.lookupObjectByParam(paramName, paramValue, objClass, true);` (or `false` depending on the default we choose)  could be readeable enough, and I think it's also shorter :-)